src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+        private const string CompletionHandlerConversion = "Conversion";
+        private const string CompletionHandlerIndexer = "Indexer";
+
+        internal override ImmutableHashSet<char> TriggerCharacters => ImmutableHashSet.Create('.');
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '.';
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static ImmutableDictionary<string, string> CreateCompletionHandlerProperty(string operation)
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, string>();
+            builder.Add(CompletionHandlerPropertyname, operation);
+            return builder.ToImmutable();
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            if (syntaxTree is { })
+            {
+                var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+                if (token.IsKind(SyntaxKind.DotToken))
+                {
+                    var expression = GetExpressionOfInvocation(token);
+                    if (expression is { })
+                    {
+                        var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
+                        var container = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+                        if (container is { })
+                        {
+                            var allMembers = container.GetMembers();
+                            var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
+                                                         where
+                                                             m.IsConversion() && // MethodKind.Conversion
+                                                             m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
+                                                             m.Parameters[0].Type == container // Convert from container type to other type
+                                                         select SymbolCompletionItem.CreateWithSymbolId(
+                                                             displayText: $"({m.ReturnType.ToMinimalDisplayString(semanticModel, position)})", // The type to convert to
```

Comment:
will def be good to have tests with interesting cases.  i.e. generic types, arrays, etc.

Reply:
I added tests for conversions to gernic types and arrays in cc4ee58a. Looks good to me.